vlm/src/main/scala/geotrellis/contrib/vlm/RasterSource.scala

```diff
@@ -41,144 +42,176 @@ import geotrellis.util.GetComponent
-        read(extent, (0 until bandCount))
-
-    /**
-      * @group read
-      */
-    def read(bounds: GridBounds): Option[Raster[MultibandTile]] =
-        read(bounds, (0 until bandCount))
-
-    /**
-      * @group read
-      */
-    def read: Option[Raster[MultibandTile]] =
-        read(extent, (0 until bandCount))
-
-    /**
-      * @group read
-      */
-    def read(bands: Seq[Int]): Option[Raster[MultibandTile]] =
-        read(extent, bands)
-
-    /**
-      * @group read
-      */
-    def readExtents(extents: Traversable[Extent], bands: Seq[Int]): Iterator[Raster[MultibandTile]] =
-        extents.toIterator.flatMap(read(_, bands).toIterator)
-
-    /**
-      * @group read
-      */
-    def readExtents(extents: Traversable[Extent]): Iterator[Raster[MultibandTile]] =
-        readExtents(extents, (0 until bandCount))
-    /**
-      * @group read
-      */
-    def readBounds(bounds: Traversable[GridBounds], bands: Seq[Int]): Iterator[Raster[MultibandTile]] =
-        bounds.toIterator.flatMap(read(_, bands).toIterator)
-
-    /**
-      * @group read
-      */
-    def readBounds(bounds: Traversable[GridBounds]): Iterator[Raster[MultibandTile]] =
-        bounds.toIterator.flatMap(read(_, (0 until bandCount)).toIterator)
-
-    /**
-     * Applies the given [[LayoutDefinition]] to the source data producing a [[LayoutTileSource]].
-     * In order to fit to the given layout, the source data is resampled to match the Extent
-     * and CellSize of the layout.
-     *
-     */
-    def tileToLayout(layout: LayoutDefinition, resampleMethod: ResampleMethod = NearestNeighbor): LayoutTileSource =
-      LayoutTileSource(resampleToGrid(layout, resampleMethod), layout)
-
-    def close = { }
+  def resampleMethod: Option[ResampleMethod]
+  def uri: String
+  def crs: CRS
+  def bandCount: Int
+
+  def cellType: CellType
+
+  /** Cell size at which rasters will be read when using this [[RasterSource]]
+    *
+    * Note: some re-sampling of underlying raster data may be required to produce this cell size.
+    */
+  def cellSize: CellSize = rasterExtent.cellSize
+
+  def rasterExtent: RasterExtent
+
+  /** All available resolutions for this raster source
+    *
+    * <li> For base [[RasterSource]] instance this will be resolutions of available overviews.
+    * <li> For reprojected [[RasterSource]] these resolutions represent an estimate where
+    *      each cell in target CRS has ''approximately'' the same geographic coverage as a cell in the source CRS.
+    *
+    * When reading raster data the underlying implementation will have to sample from one of these resolutions.
+    * It is possible that a read request for a small bounding box will results in significant IO request when the target
+    * cell size is much larger than closest available resolution.
+    *
+    * __Note__: It is expected but not guaranteed that the extent each [[RasterExtent]] in this list will be the same.
+    */
+  def resolutions: List[RasterExtent]
```

Comment:
- added comment here
Backtracking on my previous though. The `extent` here is not "useful" but `RasterExtent` nicely packs the pixel dimensions of the overviews, which we can't really get otherwise. That makes it the closest useful thing :/

Reply:
Is there smth better rather than resolutions?